src/clj/puppetlabs/services/jruby/jruby_puppet_internal.clj

```diff
@@ -250,3 +279,20 @@
         (.putFirst pool instance)))
     ;; if we get here, we got a Retry, so we just put it back into the pool.
     (.putFirst (:pool instance) instance)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
```

Comment:
I think there are some other functions in this namespace which are called from outside of this namespace, e.g., `create-pool-from-config` appears to be called from `jruby-puppet-agents`.  I like the idea of adding the "public" section here, but I think it would be good to move any of the other functions which effectively are public down into this new section.